restapi/authors.go

```diff
@@ -37,24 +36,14 @@ func (api api) CreateAuthors(w http.ResponseWriter, r *http.Request) render.Rend
 }
 
 func (api api) ReturnAuthorOriginCountries(w http.ResponseWriter, r *http.Request) render.Renderer {
-	var authors []models.Author
-	countries := make(map[string]bool)
-
-	err := api.database.Find(&authors).Error
-	if err != nil {
-		return rest.ErrInternal(api.logger, err)
-	}
+	var countries []string
 
-	for _, entry := range authors {
-		if entry.OriginCountry != "" {
-			countries[entry.OriginCountry] = true
-		} else {
-			api.logger.Debug("Empty origin country")
-		}
-	}
+    err := api.database.Table("stories").Where("is_visible = true").Distinct().Pluck("author_country", &countries).Error
```

Comment:
Does the `Distinct()` call make sure there are no duplicate countries here?

Reply:
Yeah the equivalent query is `SELECT DISTINCT author_country from stories where is_visble=true` (https://gorm.io/docs/advanced_query.html#Pluck) but i agree the test should check for this